src/engine/Account-Imap.cpp

```diff
@@ -0,0 +1,1186 @@
+/********************************************************************\
+ * Account-Imap.c -- Account Import Mapping                         *
+ * Copyright (C) 1997 Robin D. Clark                                *
+ * Copyright (C) 1997-2003 Linas Vepstas <linas@linas.org>          *
+ * Copyright (C) 2007 David Hampton <hampton@employees.org>         *
+ * Copyright (C) 2016 Robert Fewell                                 *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *                                                                  *
+\********************************************************************/
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <config.h>
+#include "AccountP.h"
+#include <qof.h>
+#include <qofinstance-p.h>
+#include "gnc-features.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+#include "Account-Imap.h"
+#include <kvp_frame.hpp>
+
+#include <iostream>
+#include <string>
+
+static QofLogModule log_module = GNC_MOD_ACCOUNT;
+
+
+/* ================================================================ */
+/* The following functions are used by
+ * src/import-export/import-backend.c to manipulate the contra-account
+ * matching data. See src/import-export/import-backend.c for explanations.
+ */
+
+GncImportMatchMap::GncImportMatchMap(Account *acc)
+{
+std::cout << "Create Map" << std::endl;
+    /* Cache the book for easy lookups; store the account/book for
```

Comment:
Don't spew to the console. Continue to use the existing ENTER and LEAVE macros, which write to gnucash.trace.